File: python/2652.py
class Solution:
    def sumOfMultiples(self, n: int) -> int:
        def s(m: int) -> int:
            k = n // m
            return k * (k + 1) // 2 * m
        return s(3) + s(5) + s(7) - s(15) - s(21) - s(35) + s(105)

File: go/2652.go
package main

func sumOfMultiples(n int) int {
	var s func(int) int
	s = func(m int) int {
		return n / m * (n/m + 1) / 2 * m
	}
	return s(3) + s(5) + s(7) + s(105) - s(15) - s(21) - s(35)
}
